srv/src/test/java/my/bookshop/CatalogServiceTest.java

```diff
@@ -0,0 +1,152 @@
+package my.bookshop;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.stream.Stream;
+
+import com.sap.cds.ql.Select;
+import com.sap.cds.services.ServiceException;
+import com.sap.cds.services.cds.CdsService;
+import com.sap.cds.services.draft.DraftService;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import cds.gen.catalogservice.AddReviewContext;
+import cds.gen.catalogservice.CatalogService_;
+import cds.gen.catalogservice.Reviews;
+import cds.gen.reviewservice.ReviewService_;
+
+@ExtendWith(MockitoExtension.class)
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+public class CatalogServiceTest {
+
+    @Autowired
```

Comment:
Can you please fix the indentation?

Reply:
Weird... locally it does not look like this... I have to make changes here, anyway, and then I'll reformat.